rt/rs/microprofile-client/src/main/java/org/apache/cxf/microprofile/client/MicroProfileClientConfigurableImpl.java

```diff
@@ -42,6 +43,8 @@
         MessageBodyWriter.class, MessageBodyReader.class, ResponseExceptionMapper.class};
     private static final String CONFIG_KEY_DISABLE_MAPPER = "microprofile.rest.client.disable.default.mapper";
 
+    private Instantiator instantiator = CDIFacade.getInstantiator().orElse(super.getInstantiator());
```

Comment:
A bit more thoughts on that, the MicroProfileClientConfigurableImpl is created from `CxfTypeSafeClientBuilder`, which it turn is created from:
 - CDI: `RestClientBean`
 - Non-CDI: `CxfRestClientBuilderResolver`
May be we could pass the instantiator through `CxfTypeSafeClientBuilder`? Fe `RestClientBean` could rely on CDI one while CxfRestClientBuilderResolver could use default. 

WDYT?

Reply:
I think it is a good idea, but when I tried it out, I realized that it breaks one of the TCK test cases - it fails when a new client is built using the `RestClientBuilder` from within a CDI-managed bean - specifically,[ this test](https://github.com/eclipse/microprofile-rest-client/blob/24096f2df248938fc00185a587bc812d969b5845/tck/src/main/java/org/eclipse/microprofile/rest/client/tck/cditests/CDIManagedProviderTest.java#L91).

Reply:
Gotcha, thanks !